app/views/Project/Tagging/Sources/SourcesFilter/index.tsx

```diff
@@ -24,61 +24,232 @@ import {
-        emmRiskFactors: [],
-        emmKeywords: [],
-        emmEntities: [],
+        entriesFilterData: [],
     }),
 };
 
-const initialValue: FilterFormType = {};
+const initialValue: FormType = {};
+
+const SOURCE_FILTER_OPTIONS = gql`
+    query SourceFilterOptions(
+        $projectId: ID!,
+    ) {
+        sourceStatusOptions: __type(name: "LeadStatusEnum") {
+            name
+            enumValues {
+                name
+                description
+            }
+        }
+        sourceExistsOptions: __type(name: "LeadExistsEnum") {
+            name
+            enumValues {
+                name
+                description
+            }
+        }
+        sourcePriorityOptions: __type(name: "LeadPriorityEnum") {
+            name
+            enumValues {
+                name
+                description
+            }
+        }
+        sourceConfidentialityOptions: __type(name: "LeadConfidentialityEnum") {
+            name
+            enumValues {
+                name
+                description
+            }
+        }
+        project(id: $projectId) {
+            members {
+                id
+                displayName
+                firstName
+                lastName
+            }
+        }
+        organizationTypes {
+            results {
+                id
+                title
+            }
+        }
+        emmEntititiesOptions: __type(name: "EmmEntityType") {
+            name
+            enumValues {
+                name
+                description
+            }
+        }
+        emmRiskFactorsOptions: __type(name: "EmmKeyRiskFactorType") {
+            name
+            enumValues {
+                name
+                description
+            }
+        }
+        emmKeywordsOptions: __type(name: "EmmKeyWordType") {
+            name
+            enumValues {
+                name
+                description
+            }
+        }
+        entryTypeOptions: __type(name: "EntryTagTypeEnum") {
+            name
+            enumValues {
+                name
+                description
+            }
+        }
+        commentStatusOptions: __type(name: "EntryFilterCommentStatusEnum") {
+            name
+            enumValues {
+                name
+                description
+            }
+        }
+    }
+`;
+
+const userLabelSelector = (
+    d: NonNullable<SourceFilterOptionsQuery['project']>['members'][number],
+) => d.displayName ?? `${d.firstName} ${d.lastName}`;
+const userKeySelector = (
+    d: NonNullable<SourceFilterOptionsQuery['project']>['members'][number],
+) => d.id;
+const organizationTypeKeySelector = (
+    d: NonNullable<NonNullable<NonNullable<SourceFilterOptionsQuery['organizationTypes']>>['results']>[number],
+) => d.id;
+const organizationTypeLabelSelector = (
+    d: NonNullable<NonNullable<NonNullable<SourceFilterOptionsQuery['organizationTypes']>>['results']>[number],
+) => d.title;
+
+function getProjectSourcesQueryVariables(
```

Comment:
It would be a lot better if we didn't need to do this transformation.

Can we generalize the actions we do there?